App/Services/EventHandlers/TextileNodeEventHandler.ts

```diff
@@ -10,112 +9,108 @@ import { contactsActions } from '../../features/contacts'
-      this.store.dispatch(TextileEventsActions.appStateChange(payload.previousState, payload.newState))
-    })
-    this.events.addListener('updateProfile', () => {
-      this.store.dispatch(TextileEventsActions.updateProfile())
-    })
-    this.events.addListener('error', (payload) => {
-      this.store.dispatch(TextileEventsActions.newErrorMessage(payload.type, payload.message))
-    })
-    // Account actions
-    this.events.addListener('setRecoveryPhrase', (payload) => {
-      this.store.dispatch(AccountActions.setRecoveryPhrase(payload.recoveryPhrase))
-    })
-    this.events.addListener('walletInitSuccess', () => {
-      this.store.dispatch(AccountActions.initSuccess())
-    })
+        // create a local log line for the threadUpdate event
+        const message = `BlockType ${type_url} on ${update.thread}`
+        this.store.dispatch(DeviceLogsActions.logNewEvent((new Date()).getTime(), 'onThreadUpdate', message, false))
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addThreadAddedListener((threadId) => {
+        this.store.dispatch(PhotoViewingActions.threadAddedNotification(threadId))
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addThreadRemovedListener((threadId) => {
+        this.store.dispatch(PhotoViewingActions.threadRemoved(threadId))
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addNotificationReceivedListener((notification) => {
+        this.store.dispatch(NotificationActions.newNotificationRequest(toTypedNotification(notification)))
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addNodeStartedListener(() => {
+        this.store.dispatch(TextileEventsActions.nodeStarted())
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addNodeStoppedListener(() => {
+        this.store.dispatch(TextileEventsActions.nodeStopped())
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addNodeOnlineListener(() => {
+        this.store.dispatch(TextileEventsActions.nodeOnline())
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addNodeFailedToStartListener((error) => {
+        this.store.dispatch(TextileEventsActions.nodeFailedToStart(error))
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addNodeFailedToStopListener((error) => {
+        this.store.dispatch(TextileEventsActions.nodeFailedToStop(error))
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addWillStopNodeInBackgroundAfterDelayListener((delay) => {
+        this.store.dispatch(TextileEventsActions.stopNodeAfterDelayStarting(delay))
+      })
+    )
+    this.subscriptions.push(
+      Textile.events.addCanceledPendingNodeStopListener(() => {
+        this.store.dispatch(TextileEventsActions.stopNodeAfterDelayCancelled())
+      })
+    )
   }
 
   tearDown() {
-    this.events.removeAllListeners()
+    for (const subscription of this.subscriptions) {
+      subscription.cancel()
```

Comment:
Event subscriptions return an instance of something called `EventSubscription` on which you can cancel the subscription later.